File: storage_sdk/src/main/java/com/walker/storage/room/viewmodel/WordViewModel.java
package com.walker.storage.room.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.walker.storage.room.model.Word;
import com.walker.storage.room.repository.WordRepository;

import java.util.List;

/**
 * Created by walkerzpli on 2021/8/5.
 */
public class WordViewModel extends AndroidViewModel {

    private final WordRepository mRepository;

    private final LiveData<List<Word>> mAllWords;

    public WordViewModel(Application application) {
        super(application);
        mRepository = new WordRepository(application);
        mAllWords = mRepository.getAllWords();
    }

    public LiveData<Word> getLastWord() {
        return mRepository.getLastWord();
    }

    public LiveData<List<Word>> getAllWords() {
        return mAllWords;
    }

    public void insert(Word word) {
        mRepository.insert(word);
    }

    public void delete(String word) {
        mRepository.delete(word);
    }

}

File: storage-ktx/src/main/java/com/demo/storage_ktx/viewmodel/WordViewModel.kt
package com.demo.storage_ktx.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.asLiveData
import androidx.lifecycle.viewModelScope
import com.demo.storage_ktx.model.Word
import com.demo.storage_ktx.repository.WordRepository
import kotlinx.coroutines.launch

/**
 * Created by lizhiping on 2023/6/5.
 * <p>
 * description
 */
class WordViewModel(private val repository: WordRepository): ViewModel() {

    val allWords: LiveData<List<Word>> = repository.allWords.asLiveData()

    fun insert(word: Word) = viewModelScope.launch {
        repository.insert(word)
    }
}

@Suppress("UNCHECKED_CAST")
class WordViewModelFactory(private val repository: WordRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(WordViewModel::class.java)) {
            return WordViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}